src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -602,15 +602,44 @@ private function registerWorkflowConfiguration(array $config, ContainerBuilder $
                 @trigger_error(sprintf('The "type" option of the "framework.workflows.%s" configuration entry must be defined since Symfony 3.3. The default value will be "state_machine" in Symfony 4.0.', $name), E_USER_DEPRECATED);
             }
             $type = $workflow['type'];
+            $workflowId = sprintf('%s.%s', $type, $name);
 
+            // Create transitions
             $transitions = array();
-            foreach ($workflow['transitions'] as $transition) {
+            $guardsConfiguration = array();
+            // Global transition counter per workflow
+            $transitionCounter = 0;
+            foreach (array_values($workflow['transitions']) as $number => $transition) {
```

Comment:
$number is unused, isn'it ?

Reply:
Indeed, it was in the previous PR. I fixed it